libsql/src/sync.rs

```diff
@@ -75,4 +99,46 @@ impl SyncContext {
     pub(crate) fn durable_frame_num(&self) -> u32 {
         self.durable_frame_num
     }
+
+    async fn write_metadata(&mut self) -> Result<()> {
+        let path = format!("{}-info", self.db_path);
+
+        let contents = serde_json::to_vec(&MetadataJson {
+            version: METADATA_VERSION,
+            durable_frame_num: self.durable_frame_num,
+        })
+        .unwrap();
+
+        tokio::fs::write(path, contents).await.unwrap();
+
+        Ok(())
+    }
+
+    async fn read_metadata(&mut self) -> Result<()> {
+        let path = format!("{}-info", self.db_path);
+
+        if !std::fs::exists(&path).unwrap() {
+            tracing::debug!("no metadata info file found");
+            return Ok(());
+        }
+
+        let contents = tokio::fs::read(&path).await.unwrap();
+
+        let metadata = serde_json::from_slice::<MetadataJson>(&contents[..]).unwrap();
+
+        assert_eq!(
+            metadata.version, METADATA_VERSION,
+            "Reading metadata from a different version than expected"
+        );
+
+        self.durable_frame_num = metadata.durable_frame_num;
+
+        Ok(())
+    }
+}
+
+#[derive(serde::Serialize, serde::Deserialize)]
+struct MetadataJson {
```

Comment:
Maybe we could add a field with hash of other fields and verify on read that file is not corrupted?

Reply:
https://github.com/tursodatabase/libsql/pull/1830/commits/de24d8211e4dd42833f962de0188087f896206f1